### PR TITLE
Added RTCInboundRTPStreamStats sample counters.

### LINKED ISSUE #152: Stat for how many audio stream packets are expanded when packets are lost (and lost and the user is speaking)
The non-standardized Chromium getStats contains the following stat:
```
// Fraction of packets that are expanded (synthesized) when we detect
// that the user is speaking.
ssrc.googSpeechExpandRate
```
What exactly does this mean? Is this useful? Should we standardize it or something similar? Sums and counters are preferred over rates.
RTCInboundRTPStreamStats already have various packet counters. Should we add another `unsigned long packetsSpeechExpanded`?

### PR DESCRIPTION
**Pull request by henbos**
Including
```
    unsigned long long totalSamples;
    unsigned long long concealedSamples;
    unsigned long long concealedAudibleSamples;
    unsigned long long concealmentEvents;
```

Fixes #152.

### CODE REVIEW

File: webrtc-stats.html [hunk 1]
```diff
@@ -1202,6 +1202,10 @@
              double              audioLevel;
              double              echoReturnLoss;
              double              echoReturnLossEnhancement;
+             unsigned long long  totalSamples;
```
Comment (alvestrand) [inline webrtc-stats.html]: samplesReceived? (Also add samplesSent?)

File: webrtc-stats.html [hunk 2]
```diff
@@ -1400,6 +1404,56 @@
...
+                </p>
+              </dd>
+              <dt>
+                <dfn><code>concealedAudibleSamples</code></dfn> of type <span class=
+                "idlMemberType"><a>unsigned long long</a></span>
+              </dt>
+              <dd>
+                <p>
+                  Only present for inbound audio tracks. The total number of concealed audio samples
+                  during audible portions of the stream. Audible means that the received audio is
```
Comment (alvestrand) [inline webrtc-stats.html]: No explanation of what "concealed" means here.
I think it should say "Number of samples that were played out based on synthesized data and not based on incoming data".

This shouldn't include comfort noise played out when CN is used, I think.

**Comment by alvestrand** [comment]
@jesup @hlundin can you review this? We're particularly concerned if we have concealedAudibleSamples defined clearly enough for interoperability.

File: webrtc-stats.html (hunk 1 above)
Comment (henbos) [inline webrtc-stats.html]: Done.

File: webrtc-stats.html (hunk 2 above)
Comment (henbos) [inline webrtc-stats.html]: Updated definition of "concealed" and referencing that definition (conceleadSamples) from concealedAudibleSamples. Audibility still defined the same way.

File: webrtc-stats.html [hunk 3]
```diff
@@ -1413,6 +1418,68 @@
...
+                </p>
+              </dd>
+              <dt>
+                <dfn><code>concealedAudibleSamples</code></dfn> of type <span class=
+                "idlMemberType"><a>unsigned long long</a></span>
+              </dt>
+              <dd>
+                <p>
+                  Only present for inbound audio tracks. The total number of concealed audio samples
+                  (see <a>concealedSamples</a>) that was played out during an audible portions of the
```
Comment (hlundin) [inline webrtc-stats.html]: portions -> portion

Comment (hlundin) [inline webrtc-stats.html]: portions -> portion

**Comment by vr000m** [comment]
@jesup @jan-ivar: need your feedback on `concealedAudibleSamples`, is the definition clear enough? is this something that you could implement by reading the spec? since we do not have a citable reference.

**Comment by vr000m** [comment]
@henbos Maybe another way forward is to make a new issue for concealedAudioSamples and a new PR, i.e., remove concealedAudioSamples from here and discuss this separately. 

Currently the totalConcealedSamples includes the concealedAudioSamples. So that we do not hold up the other metrics in the PR.

**Comment by jan-ivar** [comment]
`concealedAudibleSamples` seems fuzzy. What's the use-case?

**Comment by vr000m** [comment]
@henbos  will remove concealedAudibleSamples from PR and raise a separate issue for this.

**Comment by vr000m** [comment]
@jan-ivar what about the other metrics, are you okay with those. So that we can move forward with the PR.

**Comment by jan-ivar** [comment]
@padenot do these look good to you?

My only comment would be that it's not very clear that these are not present in the remote inbound/outbound dictionaries, but that should be solved by letting https://github.com/w3c/webrtc-stats/pull/191 merge first.

**Comment by henbos** [comment]
I removed concealedAudibleSamples from this PR to add it to a follow-up (follow-up not created until this has merged).

**Comment by vr000m** [comment]
@henbos this needs to be rebased because of the changes made by @jan-ivar

**Comment by jan-ivar** [comment]
Yeah the subject confused me, saying these were in RTCInboundRTPStreamStats when they're all in RTCMediaStreamTrackStats, so no conflict after all (I seem unable to rename it).

**Pull request merged**